### ISSUE TITLE
Issue with the Application protocol tab in the UI after running "Delete authentication protocol parameters" API calls

### ISSUE THREAD

**Issue by DulminiR** [wso2/product-is#21827]
**Describe the issue:**
After making the 'Delete Authentication Protocol Parameters' API call, the Identity Server UI displays a button to configure the protocol. When clicked, the UI shows the following message 'No templates available.' Additionally, on this page, the 'Next' button is unresponsive and does not function as expected.

**How to reproduce:**

1. Login to the IS console.
2. Create an Application and get the Application ID.
3. Run the following API call to Delete Authentication Protocol Parameters of the application
`curl -X 'DELETE' \
'https://localhost:9443/api/server/v1/applications/{application-id}/inbound-protocols/{application-type}' \
-H 'accept: */*' \
-H 'Authorization: Basic YWRtaW46YWRtaW4='
`

{application-type} should be one of the followings according to the created application.
`saml`
`oidc`
`passive-sts`
`ws-trust`

4. Go to the 'protocol' tab in created application.
5. Click 'Configure Protocols' button.

**Expected behavior:**
In the UI that pop up after clicking the 'Configure Protocols', there should be templates to select from in the 'Protocol Selection' section.

https://github.com/user-attachments/assets/c1de48aa-272e-4ad0-a1a5-b058eea13a67

**Environment information**
 - Product Version: IS 7.0.0
 - OS: Windows

**Comment by Thumimku** [comment]
Deleting protocol may be vaild in terms of REST API/ Resource point of view, but in Application Management view its a edge case. If a develop want to switch protocol they can create new app. 

In doc, we can marketing [this deletion procedure](https://is.docs.wso2.com/en/next/apis/application-rest-api/#tag/Inbound-Protocols-OAuth-OIDC/operation/deleteInboundOAuthConfiguration),  this is provided as first class support via UI. 

Hence I agree this is a valid bug to invest but not a critical bug so, reducing the priority.

CC: @NilukaSripalim @nilasini

**Comment by NilukaSripalim** [comment]
> Deleting protocol may be vaild in terms of REST API/ Resource point of view, but in Application Management view its a edge case. If a develop want to switch protocol they can create new app.
> 
> In doc, we can marketing [this deletion procedure](https://is.docs.wso2.com/en/next/apis/application-rest-api/#tag/Inbound-Protocols-OAuth-OIDC/operation/deleteInboundOAuthConfiguration), this is provided as first class support via UI.
> 
> Hence I agree this is a valid bug to invest but not a critical bug so, reducing the priority.
> 
> CC: @NilukaSripalim @nilasini @aaujayasena

Yes, rarely, an end user may face this problem, and we already provided the API and didn't mention any limitations in it, and +1 this is not critical but better we can analyze it and fix it.

**Comment by JayaShakthi97** [comment]
Currently, the UI support is only implemented to support setting up custom protocols only.
- The protocol data are fetched via `https://localhost:9443/api/server/v1/applications/meta/inbound-protocols?customOnly=true`. Note the `customOnly=true` param.
- Custom protocol configuring UI support has been provided via https://github.com/wso2/product-is/issues/19038
- Simply changing the filter param to get the default protocol list (`customOnly=false`) won't work as the UI will break in the Step 2. Therefore all the possible scenarios need to be evaluated properly and provide the fix.
